### ISSUE TITLE
Fully machine-readable datasets metadata

### ISSUE THREAD

**Issue by KOLANICH** [daviddiazvico/scikit-datasets#10]
I mean I should just call an api to get the datasets on topic, and their description should cointain enough information for domain-specific feature engineering.

And then a call without any manual tuning to do the automatic feature engineering and ML settings and then a call to tune hyperparams and a call for training a model (I have a framework doing the things relying on machine-readable specification of features, though it is very unfinished, but for now enough for my needs).

In other words, if I developed a ml software for some task, I may wanna test it and existing software against as much datasets where this kind software is applicable as possible in order to compare. To do this I need as many datasets as possible with metadata about columns. Detection if a model can be applied to the dataset is simple - if the dataset contains the column of the type learned by that model, the model can be applied to it.

**Comment by vnmabus** [comment]
Can you please provide an example of a dataset which does not have all the information that you require, and tell us what information you would add? We have tried to provide the API that sklearn.datasets follow, when it was possible, with some additions when the repository provided more information.

**Comment by KOLANICH** [comment]
>Can you please provide an example of a dataset which does not have all the information that you require,

for example this one: https://github.com/daviddiazvico/scikit-datasets/blob/master/skdatasets/ucr.py

>and tell us what information you would add

In more details it is described in https://github.com/openml/OpenML/issues/876 :

>So let's have more feature types:
>  * `{"type":"cyclic", "period": [60, 60, 24, 7]}` (`period` is an array, each element defines a period in counts of previous periods) - anything having some useful to the domain cyclic structure. Enables circle transform.
>  * `"survival"` - means that survival analysis methods should be applied to the column, if it is target. >Otherwise treat as a simple numerical.
>  * `{"type":"time", "base":0}` - means that it is absolute time. Inherits from `cyclic`.
>  * `"calendartime"` - enables feature engineering using calendar features like holidays, festivals, other periodical events dates like Olimpics, Football cup, annual conferences dates, TV schedules, etc...
>  * `"location"` - enables feature engineering tied to information from maps, like big circle distances, distances to city centre, city, state, distances to POIs of different types, etc
> * `"NLP"` - enables NLP feature engineering, like words embeddings and LSTM encoders
> * `"mysteryString"` - enables automatic feature extraction from strings which are not natural languages

>the ways how exactly the features are processed are implementation defined.
